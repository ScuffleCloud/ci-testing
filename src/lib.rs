pub mod i32 {
    pub fn add(a: i32, b: i32) -> i32 {
        a + b
    }

    pub fn mul(a: i32, b: i32) -> i32 {
        a * b
    }
}

pub mod f64 {
    pub fn add(a: f64, b: f64) -> f64 {
        a + b
    }

    pub fn mul(a: f64, b: f64) -> f64 {
        a * b
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_add_i32() {
        assert_eq!(i32::add(1, 2), 3);
    }

    #[test]
    fn test_mul_i32() {
        assert_eq!(i32::mul(2, 3), 6);
    }

    #[test]
    fn test_add_f64() {
        assert_eq!(f64::add(1.0, 2.0), 3.0);
    }

    #[test]
    fn test_mul_f64() {
        assert_eq!(f64::mul(2.0, 3.0), 6.0);
    }
}
