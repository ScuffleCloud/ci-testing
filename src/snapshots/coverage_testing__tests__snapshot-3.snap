---
source: src/lib.rs
expression: "test_branches(10.0, 1.0)"
snapshot_kind: text
---
1.5811388300841898
